spring-test/src/main/java/org/springframework/test/context/observation/ObservationThreadLocalTestListener.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.test.context.observation;
+
+import io.micrometer.observation.ObservationRegistry;
+import io.micrometer.observation.contextpropagation.ObservationThreadLocalAccessor;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.Conventions;
+import org.springframework.test.context.TestContext;
+import org.springframework.test.context.TestExecutionListener;
+
+/**
+ * {@code ObservationThreadLocalTestListener} is an extension of the {@link TestExecutionListener}
+ * SPI that updates the {@link ObservationThreadLocalAccessor} with the {@link ObservationRegistry}
+ * taken from the {@link ApplicationContext} present in the {@link TestContext}.
+ *
+ * This implementation is not thread-safe.
+ *
+ * @author Marcin Grzejszczak
+ * @since 6.0
```

Comment:
```suggestion
 * @since 6.1
```

I imagine this would be added in 6.1.

Reply:
We've made the whole train of changes in Micrometer Core (1.10.8), Context Propagation (1.0.3), Micrometer Tracing (1.0.7) and Reactor all in patch versions to fix bugs related to context propagation with reactor. We need this to be working in a patch release IMO. Without this change context propagation will not work properly. I would say that adding this component is a bug fix not an enhancement.